uktrade/data-hub-frontend
# Add UK Opportunities CollectionList page to investments app

## Description of change
This PR does a lot, so in general ... 

This PR adds the third tab to the investments app, which displays UK opportunities. This tab should show an unfiltered version of the CollectionList component, which lists all of the current UK opportunities. Filtering and downloading functionality should come in a later PR. This tab also adds links to the pre-existing create an opportunity functionality, as well as the opportunity details pages.

This PR also adds: 
- A refactor if the file structure between Opportunities list, and Opportunities Details to make a clear distinction between the two. 
- Amends a bug relating to the `addItemUrl` prop on the CollectionList component. The CollectionListHeader had the functionality to take the prop but the CollectionList component was missing the code to pass it down. 
- Adds a fix to the faulty pluralisation on the DataDownloadHeader. This header had some simple logic that would just add an 's' to the end of the `entityName` prop passed in, in order to make the grammar in the header correct. I've added an optional `entityNamePlural` prop you can pass in if your entity has a non-conforming name - ie Opportunity/Opportunities. I've also added storybook functionality to the basic story to demonstrate.


## Test instructions
After booting up your local, navigate to the investments app. You should see three tabs on this page now, with the final tab being UK Opportunities. When clicking this tab, you should see an unfiltered collection list of current opportunities. They should have 'updated on' metadata, along with dynamic location badges, which either show a singular location or should say "Multiple UK sites" if more than one location is available. When clicking on the title of the opportunity, you should be taken to that opportunity's details page. When clicking on the 'Add Opportunity' button in the header, you should be redirected to the `/investments/opportunities/create/` page. 

## Screenshots
### Before

![Screenshot 2021-04-20 at 15 56 22](https://user-images.githubusercontent.com/46787754/115418013-0780bd80-a1f1-11eb-9e00-2220083f857d.png)


### After

![Screenshot 2021-04-26 at 12 52 24](https://user-images.githubusercontent.com/46787754/116078265-4bab0c80-a68e-11eb-926b-5d12fc4ed8fc.png)



## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
